internal/controller/deployment_test.go

```diff
@@ -94,6 +94,9 @@ func TestConstructEnvForApp(t *testing.T) {
 
 		value     string
 		valueFrom *corev1.EnvVarSource
+
+		expectedOtelEndpointVarName string
+		otelEndpointValue           string
```

Comment:
I wonder if there is a more generic way you can test this? This isn't going to scale very well the second we start adding more than a few otel vars (of which there will be many).

Reply:
Perhaps just pass an otel values struct and then a map of the expected var name to value mapping?